src/test/java/com/wikia/webdriver/elements/mercury/old/SignupPageObject.java

```diff
@@ -61,38 +62,36 @@ public SignupPageObject(WebDriver driver) {
   }
 
   private SignupPageObject typeEmailAddress(String email) {
-    wait.forElementVisible(signupEmail);
-    signupEmail.sendKeys(email);
+    safelySetValue(signupEmail, email);
     return this;
   }
 
   private SignupPageObject typeUsername(String username) {
-    wait.forElementVisible(signupUsername);
-    signupUsername.sendKeys(username);
+    safelySetValue(signupUsername, username);
     return this;
   }
 
   private SignupPageObject typePassword(String password) {
-    wait.forElementVisible(signupPassword);
-    signupPassword.sendKeys(password);
+    safelySetValue(signupPassword, password);
     return this;
   }
 
+  private void safelySetValue(WebElement input, String value) {
```

Comment:
Thanks for suggestion @slayful! This is an old-school type of PageObject and currently it's hard to extract this method to a common one. I can do that in https://wikia-inc.atlassian.net/browse/IRIS-3963 which aims to refactor the whole "old" package to use the common style of PageObjects. I already added a comment there